plc4go/internal/opcua/EncryptionHandler.go

```diff
@@ -131,9 +131,9 @@ func (h *EncryptionHandler) decodeMessage(ctx context.Context, pdu readWriteMode
 		var message []byte
 		switch pduMessage := pdu.GetMessage().(type) {
 		case readWriteModel.OpcuaOpenResponseExactly:
-			message = pduMessage.GetMessage()
+			message = pduMessage.(readWriteModel.BinaryPayloadExactly).GetPayload()
```

Comment:
is it ensured that this is always `readWriteModel.BinaryPayloadExactly`?. Otherwise this part will panic

Reply:
There is only one case when we expect extensible payload while parsing and it is `CloseRequest` (which we construct). For received data we always parse `BinaryPayload` cause this part of message might be segmented.
The extensible payload is more of a helper for generating payloads at runtime rather than for receiving them.